client/src/utils/attributeAndParamUtils.js

```diff
@@ -444,6 +447,59 @@ const getParameterForRobotFromDB = async (robotId) => {
   return response;
 };
 
+/**
+ * @description Will initialize the ssot in local session storage
+ * @param {String} robotId Id of the robot for which we want to initialize the ssot locally
+ */
+const initSsotSessionStorage = (robotId) => {
+  getSsotFromDB(robotId)
+    .then((response) => response.json())
+    .then((data) => {
+      sessionStorage.setItem('ssotLocal', JSON.stringify(data));
+      sessionStorage.setItem('robotName', data.robotName);
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+
+  getAttributesFromDB(robotId)
+    .then((response) => response.json())
+    .then((data) => {
+      initSessionStorage('attributeLocalStorage', JSON.stringify([]));
+      sessionStorage.setItem('attributeLocalStorage', JSON.stringify(data));
+    });
+
+  getParameterFromDB(robotId)
+    .then((response) => response.json())
+    .then((data) => {
+      initSessionStorage('TaskApplicationCombinations', JSON.stringify([]));
+      sessionStorage.setItem(
+        'TaskApplicationCombinations',
+        JSON.stringify(data)
+      );
+    });
+
+  getParameterForRobotFromDB(robotId)
+    .then((response) => response.json())
+    .then((data) => {
+      initSessionStorage('parameterLocalStorage', JSON.stringify([]));
+      sessionStorage.setItem('parameterLocalStorage', JSON.stringify(data));
+    });
+
+  initSessionStorage('taskToApplicationCache', JSON.stringify({}));
+  initSessionStorage('availableApplications', JSON.stringify([]));
+  let applicationList = sessionStorage.getItem('availableApplications');
+  applicationList = JSON.parse(applicationList);
```

Comment:
Same as above, why not do it in one line as everywhere else in the code?

Reply:
I have really just copy pasted that code from the modeler page. But I think it is a good point. I changed it ✅